src/caselawclient/models/documents/__init__.py

```diff
@@ -310,6 +310,16 @@ def is_publishable(self) -> bool:
         # An empty list (which is falsy) therefore means the judgment can be published safely.
         return not self.validation_failure_messages
 
+    def check_source_is_not_version(self) -> SuccessFailureMessageTuple:
```

Comment:
I'm tempted to suggest a helper function that just takes a boolean and an error string and returns a SuccessFailureMessageTuple, which might allow us to inline the preponderence of these functions

success_failure = SuccessFailureMessageTuple(True, []])
success_failure += f(target_document.is_version, f"The document at {target_document.uri} is a specific version")

These similar signature functions could all potentially live in a different file too, namespaced off Document